app/assets/src/components/ui/controls/SearchBoxList.jsx

```diff
@@ -0,0 +1,103 @@
+import React from "react";
+import PropTypes from "prop-types";
+import CheckmarkIcon from "ui/icons/CheckmarkIcon";
+import Input from "ui/controls/Input";
+import cs from "./search_box_list.scss";
+import cx from "classnames";
+
+class SearchBoxList extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      filter: "",
+      selected: new Set(this.props.selected)
+    };
+
+    this.sortedOptions = this.sortOptions();
+  }
+
+  handleOptionClick = optionValue => {
+    let selected = new Set(this.state.selected);
+    if (selected.has(optionValue)) {
+      selected.delete(optionValue);
+    } else {
+      selected.add(optionValue);
+    }
+    this.setState({ selected }, () => {
+      return this.props.onChange && this.props.onChange(this.state.selected);
+    });
+  };
+
+  handleFilterChange = filter => this.setState({ filter });
+
+  sortOptions() {
+    // TODO(tcarvalho): review data structures to simplify this function
```

Comment:
You might be able to use `partition` from `lodash/fp`.

Reply:
Thanks for the tip. Will make it simpler.
Although the comment was more about the whole logic seeming weird.

Reply:
Actually no. Cannot use partition in this case, because I'm not just partitioning the array.